modules/handlers/greeting_handler.py

```diff
@@ -0,0 +1,111 @@
+import re
+from typing import Union
+from datetime import datetime, timezone, timedelta
+from slack_bolt.context.async_context import AsyncBoltContext
+
+from modules.models.slack_models.action_models import SlackActionRequestBody
+from modules.models.slack_models.command_models import SlackCommandRequestBody
+from modules.models.slack_models.event_models import MemberJoinedChannelEvent
+from modules.slack.blocks.new_join_blocks import (
+    new_join_immediate_welcome_blocks,
+    new_join_delayed_welcome_blocks,
+)
+from modules.utils import get_slack_user_by_id, log_to_thread, slack_team
+from modules.slack.blocks.greeting_blocks import (
+    initial_greet_user_blocks,
+    greeting_block_claimed_button,
+    greeting_block_button,
+)
+
+
+async def handle_new_member_join(
+    parsed_body: Union[MemberJoinedChannelEvent, SlackCommandRequestBody],
+    context: AsyncBoltContext,
+) -> None:
+    await context.ack()
+    user = None
+    if isinstance(parsed_body, MemberJoinedChannelEvent):
+        user = await get_slack_user_by_id(context.client, parsed_body.user)
+    elif isinstance(parsed_body, SlackCommandRequestBody):
+        user = await get_slack_user_by_id(context.client, parsed_body.user_id)
```

Comment:
I understand that you're passing in 2 different arguments, but this is somewhat weird to me. In a Member Joined Channel Event they call the user id `user` but in a Slack Command Request Body it's in `user_id`?

Probably a weirdness of the slack API.. this seems confusing enough to warrant a comment about why it's like this, but maybe it's a "me" problem

Reply:
I probably would have written it like this, now that I think about it (so it probably is just a "me" problem after all):

```python
user_id = parsed_body.user if isinstance(parsed_body, MemberJoinedChannelEvent) else parsed_body.user_id
user = await get_slack_user_by_id(context.client, user_id)
```

But that's like, just my [opinion man](https://media.tenor.com/images/dd0977cf6029c7a5cf364adc072c4ebe/tenor.gif) 😂

Reply:
The `user` vs `user_id` is due to some of the weirdness of Slack responses; the approach on getting it is good though, I'll take a look at implementing that.